account_regularization/models/account.py

```diff
@@ -0,0 +1,168 @@
+                                 "COALESCE(SUM(l.debit - l.credit), 0) "
+                                 "FROM account_account a "
+                                 "LEFT JOIN account_move_line l "
+                                 "ON (a.id=l.account_id) "
+                                 "WHERE a.type != 'view' "
+                                 "AND a.id IN (%s) "
+                                 "AND " + query + " "
+                                 "AND a.active "
+                                 "AND l.period_id in (%s) "
+                                 "GROUP BY a.id") % (acc_set, period_ids))
+
+        res = {}
+        for account_id, sum in self.env.cr.fetchall():
+            res[account_id] = round(sum, 2)
+        for id in self.ids:
+            res[id] = round(res.get(id, 0.0), 2)
+        return res
+
+
+class AccountRegularization(models.Model):
+    _name = 'account.regularization'
+    _description = 'Account Regularization Model'
+
+    name = fields.Char('Name', size=32, required=True)
+    account_ids = fields.Many2many('account.account',
+                                   'account_regularization_rel',
+                                   'regularization_id',
+                                   'account_id', 'Accounts to balance',
+                                   required=True,
+                                   domain=[('type', '=', 'view')])
+    debit_account_id = fields.Many2one('account.account',
+                                       'Result account,'
+                                       ' debit',
+                                       required=True)
+    credit_account_id = fields.Many2one('account.account',
+                                        'Result account,'
+                                        ' credit',
+                                        required=True)
+    balance_calc = fields.Selection([('date', 'Date'),
+                                     ('period', 'Periods')],
+                                    'Regularization time calculation',
+                                    required=True, default='period')
+
+    move_ids = fields.One2many('account.move',
+                               'regularization_id',
+                               'Regularization Moves')
+
+    @api.one
+    def regularize(self, date=time.strftime('%Y-%m-%d'),
+                   period=None, journal=None, date_to=None,
+                   period_ids=[]):
+        """ This method will calculate all the balances from all
+        the child accounts of the regularization
+        and create the corresponding move."""
+        #assert len(id) == 1, "One regularization at a time"
+        if not period or not journal:
+            raise Exception('No period or journal defined')
+        #regularization = self.browse(cr, uid, id)[0]
+        # Find all children accounts
+        account_ids = [x.id for x in self.account_ids]
+        tot_account_ids = self.env['account.account'].\
```

Comment:
@pedrobaeza I don't understand this one. I need to query childrens because I need the balance on each child account  and make a move on each of them with balance != 0